src/base/utils/net.cpp

```diff
@@ -38,5 +39,53 @@ namespace Utils
         {
             return !QHostAddress(ip).isNull();
         }
+
+        Subnet parseSubnet(const QString &subnetStr, bool *ok) {
+            const Subnet invalid = qMakePair(QHostAddress(), -1);
+            const Subnet subnet = QHostAddress::parseSubnet(subnetStr);
+            if (ok)
+                *ok = (subnet != invalid);
+            return subnet;
+        }
+
+        bool canParseSubnet(const QString &subnetStr)
+        {
+            bool ok = false;
+            parseSubnet(subnetStr, &ok);
+            return ok;
+        }
+
+        bool isLoopbackAddress(const QHostAddress &addr)
+        {
+            return (addr == QHostAddress::LocalHost)
+                    || (addr == QHostAddress::LocalHostIPv6)
+                    || (addr == QHostAddress(QLatin1String("::ffff:127.0.0.1")));
+        }
+
+        bool isIPInRange(const QHostAddress &addr, const QList<Subnet> &subnets)
+        {
+            QHostAddress protocolEquivalentAddress;
+            bool addrConversionOk = false;
+
+            if (addr.protocol() == QAbstractSocket::IPv4Protocol) {
+                // always succeeds
+                protocolEquivalentAddress = QHostAddress(addr.toIPv6Address());
+                addrConversionOk = true;
+            }
+            else
```

Comment:
Inconsistent `else` block. Curly braces are required because of curly braces in `if`.